backends/midi/coremidi.cpp

```diff
@@ -198,17 +199,40 @@ class CoreMIDIMusicPlugin : public MusicPluginObject {
 };
 
 MusicDevices CoreMIDIMusicPlugin::getDevices() const {
-	MusicDevices devices;
 	// TODO: Return a different music type depending on the configuration
-	// TODO: List the available devices
-	devices.push_back(MusicDevice(this, "", MT_GM));
+
+	MusicDevices devices;
+	char buffer[256];
+	ItemCount deviceCount = MIDIGetNumberOfDestinations();
+	for (ItemCount i = 0 ; i < deviceCount ; ++i) {
+		MIDIEndpointRef dest = MIDIGetDestination(i);
```

Comment:
De-duplicate this common name retrieval with a `Common::String getDeviceName(ItemCount deviceIndex) const` or `bool getDeviceName(ItemCount deviceIndex, Common::String &outName) const`?